MetadataMigration/src/test/java/org/opensearch/migrations/EndToEndTest.java

```diff
@@ -187,8 +185,8 @@ private void verifyCommandResults(MigrationItemResult result,
         assertThat(getNames(getFailedResultsByType(migratedItems.getIndexes(),
                 CreationResult.CreationFailureType.ALREADY_EXISTS)),
             containsInAnyOrder(testData.indexThatAlreadyExists));
-        assertThat(getNames(getSuccessfulResults(migratedItems.getAliases())),
-            containsInAnyOrder(testData.aliasNames.toArray(new String[0])));
+        // assertThat(getNames(getSuccessfulResults(migratedItems.getAliases())),
```

Comment:
For whatever reason I'm not seeing aliases being created on indices in 5.6 clusters, need to dig into this one, the feature seems to be fully documented in 5.6 [1].  Depending on how this is looking I might create a bug this and follow up separately.


- [1] https://www.elastic.co/guide/en/elasticsearch/reference/5.6/indices-templates.html

Reply:
Hmm, this is likely related to the changes I made to `createLegacyTemplate`

Reply:
As discussed: I'm okay with deferring this fix, with a reproducible description in a JIRA task for the time being.

Reply:
Figured it out! Needed to account for a breaking change in the Elasticsearch API, added new transformation TemplateMatchClausePattern,